src/Common/Auth/AuthUtils.php

```diff
@@ -386,8 +393,14 @@ public function updatePassword($activeUser, $targetUser, &$currentPwd, &$newPwd,
             return false;
         }
 
+        // Ensure password is long enough, if this option is on (note LDAP skips this)
+        if ((!self::useActiveDirectory()) && (!$this->testPasswordLength($newPwd))) {
```

Comment:
TODO
I think I've made a logic error in above self::useActiveDirectory() call (I need to go through that code block again in addition to use of self::useActiveDirectory()) here and below)